Doc/c-api/type.rst

```diff
@@ -118,7 +118,8 @@ The following functions and structs are used to create
 
 .. c:function:: PyObject* PyType_FromSpecWithBases(PyType_Spec *spec, PyObject *bases)
 
-   Creates and returns a heap type object from the *spec*.
+   Creates and returns a heap type object from the *spec*
+   (:const:`Py_TPFLAGS_HEAPTYPE`).
```

Comment:
Probably not needed, I don't see anything else in type.rst specifying this

Reply:
The intention behind making this explicit here is to help guide people towards what a heap allocated type actually is. By providing a pointer to `Py_TPFLAGS_HEAPTYPE` it lets them reference the part of the docs that mention its reference counting as well as cross-reference the portion in the `tp_dealloc` docs.